src/Workspaces/Remote/Core/Services/IAssetSource.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Serialization;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    /// <summary>
+    /// Provides assets given their checksums.
+    /// </summary>
+    internal interface IAssetSource
+    {
+        Task<IList<(Checksum, object)>> GetAssetsAsync(int scopeId, ISet<Checksum> checksums, ISerializerService serializerService, CancellationToken cancellationToken);
```

Comment:
IList?  not ImmutbaleArray? (note: remote calls support ImmutableArray Fine now).

Reply:
Switched to immutable.